### PR TITLE
Nob/refactor form

### PR DESCRIPTION
**Pull request by per-sephone**
- Added more detailed doc strings and type hints, removed unhelpful comments in form.py
- Refactored form.py by adding new functions and separating code into more specific functions

### CODE REVIEW

**Review by jasonandmonte — CHANGES_REQUESTED** [review]
I noticed that the `help` function and `.__doc__` attribute are not populating with the comments like I would expect. I think that the comments need to be put under the class/function header in order to have them associate correctly.

From:
```py
    """
    This function connects the knob values with the
    spinbox values for each knob/spinbox on the GUI
    """

    def connect_knob_and_spinbox_values(self, win):
```
To:
```py
    def connect_knob_and_spinbox_values(self, win):
        """
        This function connects the knob values with the
        spinbox values for each knob/spinbox on the GUI
        """
```

There is some examples on the PEP as well: https://peps.python.org/pep-0257/

Could we also add return types for the methods? I think most of them will be `-> None`.

Otherwise everything looks great!

**Comment by per-sephone** [comment]
Implemented your suggestions, waiting on second code review before merging

**Review by kaleb2 — APPROVED** [review]

**Review by jasonandmonte — APPROVED** [review]
Looks good! 👍👍

**Pull request merged**
